packages/next/lib/caching/valkey-components.js

```diff
@@ -0,0 +1,223 @@
+  deserialize,
+  getConnection,
+  getPlatformaticMeta,
+  getPlatformaticSubprefix,
+  keyFor,
+  serialize
+} from './valkey-common.js'
+
+export const CACHE_HIT_METRIC = {
+  name: 'next_components_cache_valkey_hit_count',
+  help: 'Next.js Components Cache (Valkey) Hit Count'
+}
+export const CACHE_MISS_METRIC = {
+  name: 'next_components_cache_valkey_miss_count',
+  help: 'Next.js Components Cache (Valkey) Miss Count'
+}
+export const MAX_BATCH_SIZE = 100
+
+export const sections = {
+  values: 'components:values',
+  tags: 'components:tags'
+}
+
+export class CacheHandler {
+  #config
+  #logger
+  #store
+  #prefix
+  #subprefix
+  #meta
+  #maxTTL
+  #cacheHitMetric
+  #cacheMissMetric
+
+  constructor () {
+    this.#config ??= globalThis.platformatic.config.cache
+    this.#logger ??= createPlatformaticLogger()
+    this.#store ??= getConnection(this.#config.url)
+    this.#maxTTL ??= this.#config.maxTTL
+    this.#prefix ??= this.#config.prefix
+    this.#subprefix ??= getPlatformaticSubprefix()
+    this.#meta ??= getPlatformaticMeta()
+
+    if (!this.#config) {
+      throw new Error('Please provide a the "config" option.')
+    }
+
+    if (!this.#logger) {
+      throw new Error('Please provide a the "logger" option.')
+    }
+
+    if (!this.#store) {
+      throw new Error('Please provide a the "store" option.')
+    }
+
+    if (globalThis.platformatic) {
+      this.#registerMetrics()
+    }
+  }
+
+  async get (cacheKey, _, isRedisKey) {
+    this.#logger.trace({ key: cacheKey }, 'cache get')
+
+    const key = isRedisKey ? cacheKey : this.#keyFor(cacheKey, sections.values)
+
+    let rawValue
+    try {
+      rawValue = await this.#store.get(key)
+
+      if (!rawValue) {
+        this.#cacheMissMetric?.inc()
+        return
+      }
+    } catch (e) {
+      this.#cacheMissMetric?.inc()
+      this.#logger.error({ err: ensureLoggableError(e) }, 'Cannot read cache value from Valkey')
+
+      return
+    }
+
+    let raw
+    try {
+      raw = deserialize(rawValue)
+    } catch (e) {
+      this.#cacheMissMetric?.inc()
+      this.#logger.error({ err: ensureLoggableError(e) }, 'Cannot deserialize cache value from Valkey')
+
+      // Avoid useless reads the next time
+      // Note that since the value was unserializable, we don't know its tags and thus
+      // we cannot remove it from the tags sets. TTL will take care of them.
+      await this.#store.del(key)
+
+      return
+    }
+
+    const { maxTTL: _maxTTL, meta: _meta, ...value } = raw
+
+    if (this.#maxTTL < raw.revalidate) {
+      try {
+        await this.#refreshKey(key, value)
+      } catch (e) {
+        this.#logger.error({ err: ensureLoggableError(e) }, 'Cannot refresh cache key expiration in Valkey')
+      }
+    }
+
+    value.value = Readable.toWeb(Readable.from(value.value))
+    this.#cacheHitMetric?.inc()
+    return value
+  }
+
+  async set (cacheKey, dataPromise, isRedisKey) {
+    const { value, ...data } = await dataPromise
```

Comment:
What is value here?

Reply:
It comes from Next.js. It contains `value, stale, revalidate` etc.